Spigot-Server-Patches/0739-Add-clear-boolean-to-Block-setType.patch

```diff
@@ -0,0 +1,55 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Fri, 21 May 2021 21:14:31 -0700
+Subject: [PATCH] Add clear boolean to Block#setType
+
+Fixes SPIGOT-3725: https://hub.spigotmc.org/jira/projects/SPIGOT/issues/SPIGOT-3725
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
+index 0006b3cad5fe46e50b0efeae99102f9d80276d61..ec6d99ad6a091df7b0b9e7ce154c2e89c7548c26 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlock.java
+@@ -169,8 +169,14 @@ public class CraftBlock implements Block {
+ 
+     @Override
+     public void setType(Material type, boolean applyPhysics) {
++        // Paper start
++        this.setType(type, applyPhysics, false);
++    }
++    @Override
++    public void setType(Material type, boolean applyPhysics, boolean clear) {
++        // Paper end
+         Preconditions.checkArgument(type != null, "Material cannot be null");
+-        setBlockData(type.createBlockData(), applyPhysics);
++        setBlockData(type.createBlockData(), applyPhysics, clear); // Paper
+     }
+ 
+     @Override
+@@ -180,11 +186,26 @@ public class CraftBlock implements Block {
+ 
+     @Override
+     public void setBlockData(BlockData data, boolean applyPhysics) {
++        // Paper start
++        this.setBlockData(data, applyPhysics, false);
++    }
++    @Override
++    public void setBlockData(BlockData data, boolean applyPhysics, boolean clear) {
++        // Paper end
+         Preconditions.checkArgument(data != null, "BlockData cannot be null");
+-        setTypeAndData(((CraftBlockData) data).getState(), applyPhysics);
++        setTypeAndData(((CraftBlockData) data).getState(), applyPhysics, clear); // Paper
+     }
+ 
+     public boolean setTypeAndData(final IBlockData blockData, final boolean applyPhysics) {
++        // Paper start
++        return this.setTypeAndData(blockData, applyPhysics, false);
++    }
++    public boolean setTypeAndData(final IBlockData blockData, final boolean applyPhysics, final boolean clear) {
++        if (clear) {
++            TileEntity tileEntity = world.getTileEntity(position);
++            net.minecraft.world.Clearable.a(tileEntity);
```

Comment:
You know the drill.

Reply:
Yep, knew about that, didnt include it cause wasn’t sure this was the right way to go about doing this, but if it is, I’ll add it.

Reply:
K, added this OBFHELPER